Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "HTTPS url scheme will be disabled on the API(s) of all API Management service." -ForegroundColor $([Constants]::MessageType.Warning)
+    
+    if (-not $Force)
+    {
+        Write-Host "Do you want to disable HTTPS url scheme for API(s) of all API Management services?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if($userInput -ne "Y")
+        {
+            Write-Host "HTTPS url scheme will not be disabled for API(s) of any API Management Service. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+    }
+    else
+    {
+        Write-Host "'Force' flag is provided. HTTPS url scheme will be disabled on the API(s) of API Management services without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 3] Disabling HTTPS url scheme for API(s) of all API Management services..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+    # Includes API Management services, to which, previously made changes were successfully rolled back.
+    $apiManagementServicesRolledBack = @()
+
+    # Includes API Management services that were skipped during roll back. There were errors rolling back the changes made previously.
+    $apiManagementServicesSkipped = @()
+
+
+    $validAPIManagementServiceDetails | ForEach-Object {
+        $apiManagementservice = $_
+           
+        Write-Host "Enabling HTTPS url scheme for API(s) of API Management Service: Resource ID - $($_.ResourceId), Resource Group Name - $($_.ResourceGroupName), Resource Name - $($_.ResourceName)" -ForegroundColor $([Constants]::MessageType.Warning)
+            
+        try
+        {           
+            $listOfApiRolledBack = @()
+            $listOfApiSkipped = @()
+            #$$apiManagementservice.ListOfAPIRemediated
+            $apiMgmtContext = New-AzAPIManagementContext -ResourceGroupName $_.ResourceGroupName -ServiceName $_.ResourceName
+            $apiMgmt = Get-AzAPIManagementAPI -Context $apiMgmtContext
+            $apiManagementservice.ListOfAPIRemediated  = $apiManagementservice.ListOfAPIRemediated -split ","
+            $apiMgmt | ForEach-Object {      
+                if($apiManagementservice.ListOfAPIRemediated -contains  $_.Name)
+                {
+                    Set-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId -Protocols @("Http") -ErrorAction SilentlyContinue
+                    $output = (Get-AzAPIManagementAPI -Context $apiMgmtContext -ApiId $_.ApiId  -ErrorAction SilentlyContinue).Protocols 
```

Comment:
Do we really need to execute another command to get the status? Please check if `Set-AzAPIManagementAPI` returns the state of the resource post the update.
Else, we can rely on the output of this command to determine if the command was a success/failure - Unless, you saw an issue where `Set-AzAPIManagementAPI()` succeeded, but, `Get..` returned a different output than the `Set..`.